data/modules/BreakdownServicing/BreakdownServicing.lua

```diff
@@ -248,28 +248,37 @@ local onEnterSystem = function (ship)
 	else
 		return -- Don't care about NPC ships
 	end
+
+	-- The more engineers the more likely to be saved (rule doesn't apply to cooking):
 	local saved_by_this_guy = savedByCrew(ship)
-	if (service_history.lastdate + service_history.service_period < Game.time) and not saved_by_this_guy then
+
+	-- Jump drive is getting worn and is running down
+	if (service_history.lastdate + service_history.service_period < Game.time) then
 		service_history.jumpcount = service_history.jumpcount + 1
-		if (service_history.jumpcount > max_jumps_unserviced) or (Engine.rand:Integer(max_jumps_unserviced - service_history.jumpcount) == 0) then
+	end
+
+	-- Test if the engine will actually break
+	if Engine.rand:Number() < service_history.jumpcount / max_jumps_unserviced then
+		print("Player hyperdrive is breaking down!")
+
+		if saved_by_this_guy then
```

Comment:
I don't get it. Why do we identify the savior before we know we need saving?

Reply:
Good catch, it's a remnant from the code I modified, as that (i.e. current master) tries to "save the ship" every jump, even if engine isn't busted. Which is why I think it's strange that this old code's strange behaviour hasn't been noticed before. Anyway, I'll fix it.